src/Components/Header/Header.js

```diff
@@ -62,57 +67,62 @@ export default function Header() {
             </h1>
           </div>
           <div className="text-gray-700 text-center flex items-center align-items">
-            <>
-              <Link to={ROUTES.DASHBOARD} aria-label="Dashboard">
-                <SvgPathComponent
-                  svgProps={svgProps}
-                  pathProps={homeIconProps}
-                />
-              </Link>
-              <button
-                type="button"
-                title="Sign Out"
-                onClick={() => {
-                  navigate("/login");
-                }}
-                onKeyDown={(event) => {
-                  if (event.key === "Enter") {
-                    // Handle sign out logic here
-                  }
-                }}
-              >
-                <SvgPathComponent
-                  svgProps={svgProps}
-                  pathProps={logoutIconProps}
-                />
-              </button>
-              <div className="flex flex-col items-center cursor-pointer">
-                <img
-                  src="/images/avatars/karl.jpg"
-                  alt="Profile"
-                  className="rounded-full h-8 w-8 flex"
-                  onClick={handleImageClick}
-                />
-                {isDropdownVisible && (
-                  <div className="dropdown-content absolute mt-6 flex flex-col">
-                    {links.map((link) => (
-                      <button
-                        key={link.id}
-                        onClick={() => {
-                          navigate(link.to);
-                        }}
-                        className="text-black-light"
-                      >
-                        {link.label}
-                      </button>
-                    ))}
+            {user ? (
+              <>
+                <Link to={ROUTES.DASHBOARD} aria-label="Dashboard">
+                  <SvgPathComponent
+                    svgProps={svgProps}
+                    pathProps={homeIconProps}
+                  />
+                </Link>
+                <button
+                  type="button"
+                  title="Sign Out"
+                  onClick={() => {
+                    firebase.auth().signOut();
+                    navigate("/login");
+                  }}
+                  onKeyDown={(event) => {
+                    if (event.key === "Enter") {
+                      firebase.auth().signOut();
+                    }
```

Comment:
Whether it is a click or a key down, in both cases you are expecting to sign out and navigate.
So please create a function that does both signing out and navigating, and just call it. 
You are duplicating the functionality twice right now.